app/controllers/users_controller.rb

```diff
@@ -311,7 +311,7 @@ def create
       return fail_with("login.reserved_username")
     end
 
-    if user = User.where(staged: true).find_by(email: params[:email].strip.downcase)
+    if user = User.where(staged: true).find_by_email(params[:email].strip.downcase)
```

Comment:
This line does not even make sense, absolutely recommend renaming it. 

find_by_email is not a scope so it will not filter on staged: true, so what we have here is both a bug and bugs waiting to happen. perhaps the scope should be `with_email` ... and then `User.where(staged: true).with_email('bob@bob.com').first`